mozilla_django_oidc_db/backends.py

```diff
@@ -61,6 +62,22 @@ def get_sensitive_claims_names(self) -> list:
         identifier_claim_name = getattr(self.config, self.config_identifier_field)
         return [identifier_claim_name] + self.sensitive_claim_names
 
+    def get_userinfo(self, access_token, id_token, payload):
+        """
+        Extract the user information, configurable whether to use the ID token or
+        the userinfo endpoint for this
+        """
+        if self.config.userinfo_claims_source == UserInformationClaimsSources.id_token:
+            logger.debug("Extracting user information from ID token")
+            return payload
+        elif (
+            self.config.userinfo_claims_source
+            == UserInformationClaimsSources.userinfo_endpoint
+        ):
+            logger.debug("Retrieving user information from userinfo endpoint")
+            return super().get_userinfo(access_token, id_token, payload)
+        logger.warning("No userinfo claim source defined!")
```

Comment:
so what happens if no value is configred and this method returns `None`? I suspect a hard crash because a dict (or similar) data type is expected.

Can you add a test for that case?

That would make this implementation a simple `if source_id_token: ...` with the default behaviour as the main body (so drop the `elif`). Otherwise instead of a `logger.warning`, this should crash hard with an exception (`ValueError` seems appropriate) or fall back to the moz-django-oidc default behaviour.

Reply:
I wasn't sure whether there would be other option we would want to support in the future, that's why I implemented it like this, but you're right, in this case it leaves the possibility for weird exceptions